app/src/main/java/com/trianglz/weatherapp/domain/usecases/getweatherdata/WeatherDataUseCase.kt

```diff
@@ -0,0 +1,41 @@
+package com.trianglz.weatherapp.domain.usecases.getweatherdata
+
+import com.trianglz.weatherapp.data.mappers.weather.toWeather
+import com.trianglz.weatherapp.data.models.city.CityDto
+import com.trianglz.weatherapp.domain.models.weather.Weather
+import com.trianglz.weatherapp.domain.repository.IRepository
+import com.trianglz.weatherapp.domain.utils.IUtilityManager
+import kotlinx.coroutines.async
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.coroutineScope
+import javax.inject.Inject
+
+class WeatherDataUseCase @Inject constructor(
+    private val repository: IRepository,
+    private val utilityManager: IUtilityManager
+) : IWeatherDataUseCase {
+
+    override suspend fun getWeatherData(
+        countryCode: String,
+        limit: Int
+    ): Result<List<Weather>> = coroutineScope {
```

Comment:
Very neat. But, what I'd do is have a use case for fetching the weather data for each city. And from the view model I'd loop and fetch multiple weather datas for each city.